docs/linking.md

```diff
@@ -0,0 +1,218 @@
+Slang Linking
+=============
+
+The Slang feature around libraries and linking are a *work in progress*. Future versions of Slang are likely to change the API and binary compatibility. Also note that currently reflection is mainly unsupported. 
+
+In many languages it is possible to compile source files into binaries such as libraries and object files. Later these files can be combined perhaps with other source files to produce a result such as an executable. Slang now has experimental support for such a feature. 
+
+To make such a feature work we need a few abilities
+
+* To create a library from source 
+* A way for symbols to be exported and imported
+* Linkage - ability to combine source/libraries to produce things (such as other libraries, executables, etc)
+* A way to specify libraries to be used in linkage
+
+Libraries
+---------
+
+A library can be created from source from the Slang command line with the `-o` option naming the output file. The filename currently requires a `slang-lib` or `slang-module` extension. This extension will mean the library will be stored as serialized Slang IR. The option `-no-codegen` is also typically specified because we don't want to generate any *target* code, such as dxil, spir-v and so forth when producing a library. 
+
+It is important to note that due to how the feature currently works, if it contains any kind of binding it should be considered platform specific. 
```

Comment:
This line could maybe be beefed up to be its own section, since it is more than just binding/layout stuff being target-specific, but also that if you are linking two things that both declare global shader parameters then they had better all have manually assigned locations for now (and you can't go around using entry-point `uniform` parameters).